### ISSUE TITLE
npm i fails - duplicate in path

### ISSUE THREAD

**Issue by mrchief** [ArtiomTr/jest-coverage-report-action#91]
```
Run artiomtr/jest-coverage-report-action@v2.0-rc.1
  with:
    github-token: ***
    annotations: failed-tests
    test-script: npx jest --silent --ci --coverage --testLocationInResults --json --outputFile="report.json"
    icons: emoji
    package-manager: npm
    skip-step: none
  env:
    NPM_CONFIG_USERCONFIG: /home/runner/work/_temp/.npmrc
    NODE_AUTH_TOKEN: XXXXX-XXXXX-XXXXX-XXXXX
/opt/hostedtoolcache/node/14.17.0/x64/bin/npm install
npm WARN tar ENOENT: no such file or directory, open '/home/runner/work/my-app/my-app/node_modules/.staging/styled-components-51686fc4/dist/styled-components-macro.cjs.js'
npm WARN tar ENOENT: no such file or directory, open '/home/runner/work/my-app/my-app/node_modules/.staging/styled-components-51686fc4/dist/styled-components-macro.cjs.js.map'
npm WARN tar ENOENT: no such file or directory, open '/home/runner/work/my-app/my-app/node_modules/.staging/styled-components-51686fc4/dist/styled-components-macro.esm.js'
npm WARN tar ENOENT: no such file or directory, open '/home/runner/work/my-app/my-app/node_modules/.staging/styled-components-51686fc4/dist/styled-components-macro.esm.js.map'
npm WARN tar ENOENT: no such file or directory, open '/home/runner/work/my-app/my-app/node_modules/.staging/react-c9bd9316/cjs/react-jsx-runtime.production.min.js'
npm WARN tar ENOENT: no such file or directory, open '/home/runner/work/my-app/my-app/node_modules/.staging/styled-components-51686fc4/dist/styled-components.browser.cjs.js'
...
```

I suspect it has something to do with dup in path - `my-app` appears twice `/home/runner/work/my-app/my-app/node_modules`

With `skip-step: install`, I get  
```
Error: The process '/opt/hostedtoolcache/node/14.17.0/x64/bin/npx' failed with exit code 1
    at f._setResult (/home/runner/work/_actions/artiomtr/jest-coverage-report-action/v2.0-rc.1/dist/index.js:2:11470)
    at f.CheckComplete (/home/runner/work/_actions/artiomtr/jest-coverage-report-action/v2.0-rc.1/dist/index.js:2:10900)
    at ChildProcess.<anonymous> (/home/runner/work/_actions/artiomtr/jest-coverage-report-action/v2.0-rc.1/dist/index.js:2:9944)
    at ChildProcess.emit (events.js:210:5)
    at maybeClose (internal/child_process.js:1021:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
```

The only way to make it work is if I install, test on my own but then I lose out on annotations. 

```
  run_tests:
    name: Tests
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v2

      - uses: actions/setup-node@v2
        with:
          node-version: '14.x'
          always-auth: true
          registry-url: https://npm.pkg.github.com
          scope: '@acme-corp'

      - name: Cache node_modules
        uses: actions/cache@v2
        with:
          path: '**/node_modules'
          key: ${{ runner.os }}-node_modules-${{ hashFiles('**/package-lock.json') }}

      - name: install deps
         run: npm install
         env:
           NODE_AUTH_TOKEN: ${{secrets.BOT_TOKEN}}

       - name: Run Tests & Collect Coverage
         run: npx jest --silent --ci --coverage --testLocationInResults --json --outputFile="report.json"

      - name: Report Coverage
        uses: artiomtr/jest-coverage-report-action@v2.0-rc.1
        with:
          github-token: ${{ secrets.GHPR_TOKEN }}
          skip-step: all
          annotations: failed-tests
```

Using a `GHPR_TOKEN` as I need to read private packages from github package registry.

**Comment by mrchief** [comment]
Noticed this 

```
npm WARN tar ENOENT: no such file or directory, open '/home/runner/work/my-app/my-app/node_modules/.staging/@storybook/cli-1f575d53/dist/cjs/project_types.js'
npm ERR! code E401
npm ERR! 401 Unauthorized - GET https://npm.pkg.github.com/download/@acme-corp/eslint-config/0.1.9/db9b489fc087654dad55a0d02cdd0258f62378e85fb23f8520ab24dee522dec1
```

I think this action overrides `.npmrc` with it's own?

**Comment by ArtiomTr** [comment]
Hello @mrchief,

This is a pretty silly action, it just runs the npm install command without touching the .npmrc files, etc. ([code here](https://github.com/ArtiomTr/jest-coverage-report-action/blob/cf09b23407c2c1243413387ebfa09f1d02b559af/src/collect/getRawCoverage.ts#L43-L52)). By default, dependencies are installed in cwd, so if you think duplicate path is a mistake, you can specify custom `working directory`. In my opinion, `working-directory` is set up correctly, and there is a problem with authorization to the npm registry. As [stated in the npm documentation](https://docs.npmjs.com/using-private-packages-in-a-ci-cd-workflow#set-the-token-as-an-environment-variable-on-the-cicd-server), you can specify the `NPM_TOKEN` environment variable to authorize to the private registry. As I understand, you are using the `BOT_TOKEN` secret for authorization. 

To sum up, I think your action should look like this:

```yml
- name: Report Coverage
  uses: artiomtr/jest-coverage-report-action@v2.0-rc.1
  env:
    NPM_TOKEN: ${{ secrets.BOT_TOKEN }}
  with:
    github-token: ${{ secrets.GHPR_TOKEN }}
    annotations: failed-tests
```

**Comment by mrchief** [comment]
We're using https://github.com/actions/setup-node so what eventually worked (in `npm i`) was 

```
      - name: Report Coverage
        uses: artiomtr/jest-coverage-report-action@master
        env:
          NODE_AUTH_TOKEN: ${{ secrets.BOT_TOKEN }}
        with:
          github-token: ${{ secrets.GITHUB_TOKEN }}
          annotations: failed-tests
```
I deliberately added a failing test in my PR to test the annotation and the spoiler parts in comment and I cann see it failing in logs: 
```
Test Suites: 1 failed, 3 passed, 4 total
Tests:       1 failed, 13 passed, 14 total
Snapshots:   0 total
Time:        6.794 s
Test results written to: report.json
Test execution failed with error: Error: The process '/opt/hostedtoolcache/node/14.17.0/x64/bin/npx' failed with exit code 1
    at f._setResult (/home/runner/work/_actions/artiomtr/jest-coverage-report-action/master/dist/index.js:2:11470)
    at f.CheckComplete (/home/runner/work/_actions/artiomtr/jest-coverage-report-action/master/dist/index.js:2:10900)
    at ChildProcess.<anonymous> (/home/runner/work/_actions/artiomtr/jest-coverage-report-action/master/dist/index.js:2:9944)
    at ChildProcess.emit (events.js:210:5)
    at maybeClose (internal/child_process.js:1021:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
```

but it failed to do the annotations and also, everything shows as green despite the failed test. I think that's because the tests against `main` pass so ultimately the exit code is 0. 

![image](https://user-images.githubusercontent.com/781818/123547794-2428fc80-d730-11eb-8c7e-b1e9acee5805.png)

**Comment by mrchief** [comment]
Ok, so we're getting closer - I was able to get the annotations working as well as correct status by setting `github-token` to `secrets.GITHUB_TOKEN`

```
      - name: Report Coverage
        uses: artiomtr/jest-coverage-report-action@master
        env:
          NODE_AUTH_TOKEN: ${{ secrets.BOT_TOKEN }}
        with:
          github-token: ${{ secrets.GITHUB_TOKEN }}
          annotations: failed-tests
```
It feels a little weird though becuase the failure is not on the actual "Test" but a new one called "Tests' Annotations"
![image](https://user-images.githubusercontent.com/781818/123549719-2a22db80-d738-11eb-809b-951478f8c124.png)

I still don't see the the spoiler tags in the PR comment. Not sure what I'm missing.

**Comment by ArtiomTr** [comment]
This is a pretty tricky question.

This action was originally designed only to report code coverage on a pull request. From this point of view, the action should not fail when tests fail - this action does not validate the tests.

However, when annotations for failed tests were implemented, it is counterintuitive that the failed test is annotated but action succeeds.

This is a pretty interesting topic for discussion, but unfortunately, it does not apply to your problem. I will close the issue and open the topic in discussions - feel free to ask more questions or express your opinion on this topic!

**Comment by ArtiomTr** [comment]
Opened discussion #94
